conductor.go

```diff
@@ -48,38 +44,29 @@ func (c *Conductor) Play(ctxMain context.Context) error {
 		})
 	}()
 
-	var errs = make(chan InstrumentError)
+	// Unbuffered channels would block if nothing is reading off them
+	// So buffered them to the len of Players so goroutines are not blocked
+	// if they all error.
+	var errs = make(chan InstrumentError, len(c.Players))
 	var allDone = make(chan struct{})
 
+	var wg sync.WaitGroup
 	wg.Add(len(c.Players))
 	for name, p := range c.Players {
-		go func(name string, p Player) {
-
-			// The function to play our player
-			play := func(p Player) {
-				err := p.Play(ctxWthCancel)
-				if err != nil {
-					errs <- InstrumentError{name, err}
-				}
-			}
+		_, exists := c.playing[name]
 
-			lock.RLock()
-			_, exists := c.playing[name]
-			lock.RUnlock()
+		if exists {
+			// Ensure we inform all other players to shutdown
+			cancel()
+			return fmt.Errorf("%s already exists", name)
+		}
 
-			if !exists {
-				lock.Lock()
-				c.playing[name] = struct{}{}
-				lock.Unlock()
+		go func(name string, p Player) {
+			defer wg.Done()
```

Comment:
I like the use of `defer` here, but we should move it to the top of the function.